.github/workflows/wiki-documentation.yaml

```diff
@@ -0,0 +1,36 @@
+# This workflow publishes Markdown docs to Github Wiki.
+# Some manual setup is required before using it:
+#   1. Enable Wiki in repository and create at least one page.
+#   2. Generate a Personal Access Token (PAT) with "repo" scope enabled.
+#   3. Add the generated PAT as a secret named "GH_PERSONAL_ACCESS_TOKEN".
+# For details, see https://github.com/marketplace/actions/publish-to-github-wiki#setup
+name: Generate Wiki Pages
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  generate-wiki:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.10'
+    - name: Install dependencies
+      run: make setup
+      env:
+        PYTHON: python
+    - name: Generate Markdown docs
+      run: |
+        make docs-md
+        mv docs/_build/markdown/index.md docs/_build/markdown/Home.md
+    - name: Push to Wiki
+      uses: SwiftDocOrg/github-wiki-publish-action@v1
+      with:
+        path: docs/_build/markdown
+      env:
+        GH_PERSONAL_ACCESS_TOKEN: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
```

Comment:
What permissions does this token need, so we can set it up for this repo. We currently use the standard [`GITHUB_TOKEN`](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret) and a `RELEASE_TOKEN` with `repo` permissions (see [discussion](https://github.com/relekang/python-semantic-release/issues/311#issuecomment-1157270026)).

Reply:
`GITHUB_TOKEN` can only push to the repo containing the workflow (i.e. this repo). The action `SwiftDocOrg/github-wiki-publish-action` needs to push to a separate repo where Github stores the Wiki, in this case https://github.com/jenstroeger/python-package-template.wiki (note the `.wiki` at the end). That's why [we need a PAT](https://github.com/marketplace/actions/publish-to-github-wiki#3-generate-a-personal-access-token).

We can either create yet another PAT with `repo` permissions for pushing to Wiki, or replace `RELEASE_TOKEN` with a generic PAT used for semantic release, wiki publishing and whatever else doesn't work with `GITHUB_TOKEN`.

Reply:
@lecram gotcha, the required token scope is also documented in the [`SwiftDocOrg/github-wiki-publish-action` repository](https://github.com/SwiftDocOrg/github-wiki-publish-action#setup).

I’m happy to create another `DOC_TOKEN` (or whatever appropriate name we’d like to use), or rename the existing `RELEASE_TOKEN` (which already has the correct [`repo` scope](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes)) to something more general (e.g. `REPO_TOKEN` to express the token’s scope?).

@behnazh, what are your thoughts?

Reply:
> I’m happy to create another `DOC_TOKEN` secret (or whatever appropriate name we’d like to use), or rename the existing `RELEASE_TOKEN` secret (which already contains the correct PAT with [`repo` scope](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes)) to something more general (e.g. `REPO_TOKEN` to express the token’s scope?).
> 
> @behnazh, what are your thoughts?

If the token requires the exact access levels as the `RELEASE_TOKEN`, it would make sense to create a more general `REPO_TOEKN`. However, the `repo` scope has[ several access levels](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes). What is the exact access level required for the release and wiki workflows?

Reply:
> What is the exact access level required for the release and wiki workflows?

I don’t know without trying out the different `repo:*` scopes 😳

Reply:
Looks like the wiki Action needs [all](https://github.com/marketplace/actions/publish-to-github-wiki#3-generate-a-personal-access-token) `repo` access levels :thinking:

Reply:
Thanks @behnazh. Let me add a new secret `REPO_TOKEN` that we can use here, and which needs to replace the `RELEASE_TOKEN` introduced with PR https://github.com/jenstroeger/python-package-template/pull/219.

Reply:
How about `REPO_ACCESS_TOKEN`, see PR https://github.com/jenstroeger/python-package-template/pull/231?

Reply:
Updated to use `REPO_ACCESS_TOKEN` in 65ef5a8b5529bc826927b3504d7f0ba6e848ac52.